src/ipaperftest/plugins/apitest.py

```diff
@@ -88,8 +92,10 @@ def run(self, ctx):
                 r"| at {time}".format(
                  cmd=formated_api_cmd, id=str(i), time=local_run_time)
             )
+            commands[client_idx].append(cmd)
+        for id, command_list in enumerate(commands):
             sp.run(
-                'vagrant ssh {} -c "{}"'.format(clients[client_idx], cmd),
+                'vagrant ssh {} -c "{}"'.format(clients[id], " && ".join(command_list)),
```

Comment:
If any command fails the remaining will also fail, is that desired? Perhaps semi-colon instead of &&?

Reply:
These commands shouldn't fail. Note that they are not the actual client installation, just scheduling via `at`.

Reply:
What is amount? Won't this do multiple kinit, one per command? That is likely slowing things down as well (and can fail on a busy server).
So if I have user-add with an amount of 500 this would create 500 of these invocations?